zcash_client_backend/src/keys.rs

```diff
@@ -536,19 +537,46 @@ impl UnifiedFullViewingKey {
         self.orchard.as_ref()
     }
 
-    /// Attempts to derive the Unified Address for the given diversifier index.
+    /// Attempts to derive the Unified Address for the given diversifier index and
+    /// receiver types.
     ///
     /// Returns `None` if the specified index does not produce a valid diversifier.
-    // TODO: Allow filtering down by receiver types?
-    pub fn address(&self, j: DiversifierIndex) -> Option<UnifiedAddress> {
-        let sapling = if let Some(extfvk) = self.sapling.as_ref() {
+    pub fn address(
+        &self,
+        j: DiversifierIndex,
+        receiver_types: &[Typecode],
```

Comment:
This change should maybe be documented, although we did introduce this type as experimental so maybe we just change them unannounced.